cleanup interface for setting multiple audio-source volumes

MaZderMind opened the pull request:
I tried to reverse-engeneer the format of the `audiosource` with multiple
sources and their volumes and found it to be "source:volume,source:volume"

With the merge of #110 we now have a way of specifying per-source properties.
This PR simplifies the configuration, so that
a) audiosource can only old one source which is a shorthand of setting the
   specified source to volume=1.0
b) sources volume's are now specified as `[source.name] volume=1.00` and this is
   the only way of specifying multiple audio-sources

I know this is a backward incompatible change, but moving towards a 1.0 I would
like to clean up the interface (which is mostly made out of the configuration)
to be as surprise-free as possible.

Also this new format would fit nicely with #104, where we could specify a
`[source.streamblanker] volume=0.5` in the same fashion.

This is a suggestion and I would like to hear some feedback, esp. from @Florob
and @DavidVentura who seemd to be involved in this feature.

Florob commented on voctocore/lib/audiomix.py:
```diff
@@ -7,18 +9,33 @@
+            try:
+                volume = Config.getfloat(section, 'volume')
+                self.log.info('Configuring Mixer-Pad %s (%u) to %f',
+                              name, index, volume)
```
Same as ll. 25-26.

Florob commented on voctocore/lib/audiomix.py:
```diff
@@ -7,18 +9,33 @@
+                self.log.info('Configuring Mixer-Pad %s (%u) to %f',
+                              name, index, volume)
+                self.volumes[index] = volume
+            except (NoSectionError, NoOptionError):
```
It might be better to just have a fallback of `0.0`. This will also give us a log message for each source, instead of skipping those without configuration. OTOH, maybe there is some benefit to only logging those with configuration.

MaZderMind commented:
@Florob thank you for you feedback. I'll work on the inline-comments, I am ok with all of them.

Some ideas regarding the aformentioned interactions:
> audiosource doubles as "don't let voctogui control the volume", which to me implies that it configures the one volume setting you always want (i.e. not necessarily just a startup default). I'm not too fond of this, but we should probably talk about expectations people have for this. We should also decide if per source volume configuration should have similar behavior.
I would like to implement the audiosource-Option to truly be a shortcut for setting the volume on one of the channels. This implies that when a volume has been configured for any channel, the UI will not show volume-controls unless forced to (via `[misc] forcevolumecontrol`).
This is close to the current behavior: configuring the volume disables dynamic changes.

> having audiosource and per source configuration allows them to be at odds. Worst case is probably audiosource = cam1 and [source.cam1] volume = 0.0
With `audiosource` being just a shortcut I'd be happy to raise a RuntimeException (or even better introduce a ConfigurationException) when the `audiosource`-shortcut is being used together with per-channel volume configurations.

MaZderMind commented:
@Florob I changed the behavior as described above and also modified the log-messages and the behavior with unknown source-names.

I also added a first unit-test which tests `Audiomix`s behavior in the different kind of configurations and added it to the travis-build.

MaZderMind commented:
The changes made to `voctocore/config.py` and `voctocore/args.py` whould be made to voctogui, too, but this PR already has a log of things in it that don't really belong here, so I opened another Issue for 1.0 to make them shared: #146

Florob commented:
`mock` is part of Python since Python 3.3 as `unittest.mock`. In practice we rely on 3.4 anyway due to our use of `enum`.

For comments my personal preference would be `#` for documentation, and `;` for options, but no strong opinion there.

MaZderMind commented:
@Florob As mentioned on IRC I would like to split this PR into two halfs: The introduction of the unit-tests and on top of that this feature, even if it's only for the cleaner git history.

MaZderMind commented:
this is now based on #147 which does contain the unit-test logic (while this pr only adds the audio parsing logic + the required test).

MaZderMind commented:
merged the unit-test PR and rebased this ontop of the resulting master. this PR now only contains the inteface cleanup.

MaZderMind commented:
I also rebased the ini-file changes out. does not belong here.
